config/routes.rb

```diff
@@ -43,14 +43,18 @@ def put(*_args); end
         post :copy, :action => 'copy', :controller => 'portfolios'
         post :undelete, :action => 'restore', :controller => 'portfolios'
       end
+      post '/portfolio_items/:portfolio_item_id/icon', :action => 'add_icon_to_portfolio_item', :controller => 'portfolio_items'
```

Comment:
Do we need the `:controller` here? Possibly we should be using the rails shortened form https://guides.rubyonrails.org/v5.1/routing.html

`post '/portfolio_items/:portfolio_item_id', :to => 'portfolio_items#add_icon_to_portfolio_item'`

Reply:
Oh cool, I like that syntax! Didn't know about it and was just following the rest of the routes. Will use this from now on.